protocol-jackson/src/main/java/io/camunda/zeebe/protocol/jackson/record/LazyJsonSerializable.java

```diff
@@ -0,0 +1,23 @@
+package io.camunda.zeebe.protocol.jackson.record;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.protocol.record.JsonSerializable;
+import java.io.UncheckedIOException;
+
+interface LazyJsonSerializable extends JsonSerializable {
+  /** @deprecated use your own {@link ObjectMapper} instead of relying on this one */
+  @Deprecated(since = "1.2.0")
+  ObjectMapper JSON_WRITER = new ObjectMapper();
+
+  /** @deprecated use a {@link ObjectMapper} directly instead of calling this method */
+  @Override
+  @Deprecated(since = "1.2.0")
+  default String toJson() {
+    try {
+      return JSON_WRITER.writeValueAsString(this);
+    } catch (final JsonProcessingException e) {
+      throw new UncheckedIOException(String.format("Failed to serialize %s to JSON", this), e);
+    }
+  }
```

Comment:
I don't really see how this default implementation is lazy. 

🔧 Please either keep track of the first result and return it on each consecutive call; or rename the interface to something like `SimpleJsonSerializable` or `DefaultJsonSerializable` or something else.